GrindingCity.WebApi/Controllers/ResourseController.cs

```diff
@@ -7,52 +8,60 @@ namespace GrindingCity.WebApi.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
-    public class ResourseController : Controller
+    public class ResourceController : Controller
     {
         // GET all api
         [HttpGet()]
-        public async Task<ActionResult<IEnumerable<Resourse>>> GetAllResourses(
-            [FromServices] IResourseRepository handler)
+        public async Task<ActionResult<IEnumerable<Resource>>> GetAllResources(
+            [FromServices] IResourceRepository resourceHandler)
         {
-            var result = await handler.GetAllResourses();
+            var result = await resourceHandler.GetAllResourcesAsync();
 
             return result is null ? NotFound() : Ok(result);
         }
 
         // GET api
         [HttpGet("{id:guid}")]
-        public async Task<ActionResult<Resourse>> GetResourse(Guid id,
-            [FromServices] IResourseRepository handler)
+        public async Task<ActionResult<Resource>> GetResource(Guid id,
+            [FromServices] IResourceRepository resourceHandler)
         {
-            var result = await handler.GetResourse(id);
+            var result = await resourceHandler.GetResourceAsync(id);
 
             return result is null ? NotFound() : Ok(result);
         }
 
         // POST api
         [HttpPost]
-        public async Task<ActionResult> AddResourse([FromForm] CreateResourseRequest resourse,
-            [FromServices] IResourseRepository handler)
+        public async Task<ActionResult> AddResource([FromForm] CreateResourceRequest request,
+            [FromServices] IResourceRepository resourceHandler,
+            [FromServices] IBuildingRepository buildingHandler)
         {
-            await handler.AddResourse(resourse);
+            _ = await buildingHandler.GetBuildingAsync(request.buildingId) 
+                ?? throw new InvalidBuidingException();
+
+            await resourceHandler.AddResourceAsync(request);
             return NoContent();
         }
 
         // PUT api
         [HttpPut("{id:guid}")]
-        public async Task<ActionResult> UpdateResourse(Guid id, [FromBody] UpdateResourseRequest resourse,
-            [FromServices] IResourseRepository handler)
+        public async Task<ActionResult> UpdateResource(Guid id, [FromBody] UpdateResourceRequest request,
+            [FromServices] IResourceRepository resourceHandler,
+            [FromServices] IBuildingRepository buildingHandler)
         {
-            await handler.UpdateResourse(id, resourse);
+            _ = await buildingHandler.GetBuildingAsync(request.buildingId)

```

Comment:
why don't we handle this with http codes?

Reply:
updated, like this if (building is null) return NotFound();?